lib/src/dbs/options.rs

```diff
@@ -49,7 +51,9 @@ pub struct Options {
 	/// Should we process variable field projections?
 	pub projections: bool,
 	/// The channel over which we send notifications
-	pub sender: Option<Sender<Notification>>,
+	/// Must be set alongside live, and preferably populated via datastore notifications channel
+	/// TODO create ticket to sort this
+	pub sender: Option<Arc<RwLock<Sender<Notification>>>>,
```

Comment:
The Option structure is used everywhere, and this is now introducing a risk of deadlock. Is there an alternative?

Reply:
What are we trying to achieve by using the lock here... To be able to save the `Sender<Notification>` later in a readonly `sender`?

Reply:
Because realtime is optional. We also have Arc RwLock because it needs to be Send.

We could have deadlocks, that's true. Gotta be careful with how we lock things, and remember to immediately release. Not sure if there is an immediate workaround, even with Pin, since we need the RwLock I believe.

@rushmorem yes, the executor takes a session which it uses to populate the options sent into the query handling. All these statements (live.rs) and docs (lives.rs) need access to the channel.

Reply:
@phughk In that case we should be able to use `std::sync::OnceLock` here instead of a `RwLock`, unless I'm missing something. If we do that we can drop the `Option` too since we can use `OnceLock::new()` to create an empty cell.